sys/include/net/gnrc/ipv6.h

```diff
@@ -137,6 +137,19 @@ kernel_pid_t gnrc_ipv6_init(void);
  */
 void gnrc_ipv6_demux(kernel_pid_t iface, gnrc_pktsnip_t *current, gnrc_pktsnip_t *pkt, uint8_t nh);
 
+/**
+ * @brief   Get the IPv6 header from a given list of @ref gnrc_pktsnip_t
+ *
+ *          This function may be used with e.g. a pointer to a (full) UDP datagram.
+ *
+ * @param[in] pkt    The pointer to the first @ref gnrc_pktsnip_t of the
+ *                   packet.
+ *
+ * @return A pointer to the @ref ipv6_hdr_t of the packet.
+ * @return NULL if the packet does not contain an IPv6 header.
+ */
+ipv6_addr_t *gnrc_get_ipv6_header(gnrc_pktsnip_t *pkt);
```

Comment:
To keep I'm line with the rest of the naming in GNRC I would prefer `gnrc_ipv6_get_hdr() `. 

Sidenote: why is the return-type `ipv6_addr_t` that doesn't fit the doc.